wordpress-mobile/WordPress-Android
# Stop recreating fragments on author filter changed

Fixes #9522 

We were recreating all fragments in the ViewPager when author filter selection changed. This had negative performance impact, we lost the item added/removed animation and it didn't make much sense anymore (since we support changing list's filtering options on the go).

This PR takes care of updating all the child-fragments when the author filter selections changes.

~Note: I noticed, post of other authors are displayed only on "Published" tab even when "Everyone" filter is selected. However, this was the case even before this change + it seems Calypso works the same way.~

Note2: I also fixed some minor issues - such as scroll position being lost on rotation.

To test:
1. My Site -> Post List
2. Change author filter to "Everyone" and make sure you see posts of other author
3. Change author filter to "Me" and make sure you see only your posts
4. Go to Trashed Tab and change the filter
5. Go back to Published tab and make sure you see the correct posts
6. Rotate your device
7. Make sure you see the correct posts and the scroll position wasn't reseted
8. Make sure the Search feature still works as expected


Update release notes:

- The changes are mostly technical



| ![author-filter-not-fixed](https://user-images.githubusercontent.com/2261188/61770184-f96bf600-adec-11e9-9654-ff9d24e16172.gif)  |  ![author-filter-fixed](https://user-images.githubusercontent.com/2261188/61770186-f96bf600-adec-11e9-9f04-b5eb3610bf70.gif) |
|---|---|
